R/detectStandard.R

```diff
@@ -0,0 +1,47 @@
+#' Detect the CDISC data standard used for a data set
+#'
+#' This function attempts to detect the data standard used for a given R data frame. 
+#'
+#' @param data A data frame in which to detect the data standard 
+#' @param domain The data domain for the data set provided. Currently, only "labs" is supported. Default: \code{"labs"}.
+#' @return A list with the standard ("ADaM", "SDTM" or "None") and a description of the matching results
+
+detectStandard <- function(data, domain="labs", details=FALSE){
+  # Create placeholder list, with Standard = None.
+  standard_list <- list()
+  standard_list[["details"]] = list()
+  data_cols<-colnames(data)
+  # Define Required Columns for the data standards
+  # TODO: Move to a more robust comparison, possibly with JSON Schema. 
+  
+  ADaM_cols <- c("USUBJID","AVAL","PARAM","VISIT","VISITNUM","ADY","A1LO","A1HI")
+  SDTM_cols <- c("USUBJID","STRESN","TEST","VISIT","VISITNUM","DY","STNRLO","STNRHI")
+  
+  #helper function that returns a summary of which data columns are found in a given standard
+  compare_cols<-function(data_cols, standard_cols){
```

Comment:
Would recommend defining compare_cols() separately.  Can live in the same file, just at the top, and we don't need to export to the user.